### DISCUSSION TITLE
Color / wire color

### DISCUSSION THREAD

**Discussion by PaulKappmeyer** [hneemann/Digital#812] [Ideas]
So, I was just generally curious if it is possible to set more individual colors.
Of course, I found the color scheme option in the settings menu, but this only allows me to edit the general appearance of the circuit.

It would be very nice to be able to edit for example individual wire colors or even shape colors.

**Comment by hneemann** [comment]
The only thing that is possible is to give your own embedded circuits their own color.
I don't think I will extend anything here either.

**Reply by PaulKappmeyer** [reply]
That’s a bummer :).
But would you like to explain your concerns about this feature?

Because:
1.	I think I saw this feature being requested several times before.
2.	I’m just thinking about adding/displaying colors in edit mode (so not when running the simulation; see screenshot) – so I don’t think that this feature would bring performance issues to the simulation.
![colors](https://user-images.githubusercontent.com/42696082/131504905-b406d9eb-b58e-4f36-9bf2-fdee2836670f.png)

3.	Adding a color-attribute to the wires in the .dig-file wouldn’t make the files that much larger (I guess ^^)
4.	And the required features are already there; (e.g. the “Wire tool tips” identifies a connected net of wires, and the color picker/color palette in the LED works just fine).

Thanks again for your work :-)

**Reply by hneemann** [reply]
In my opinion, the settings menus already contain too much clutter. 
I have added too many options in the past. I don't want to add more niche features that offer little functionality. 
In the end, it's a tool meant for teaching. It must remain easy to use. Otherwise it no longer meets its primary purpose.
I don't want the students tinkering around with wire colors and all sorts of things and losing control of the many different settings.

It also adds complexity without adding much functionality.

**Reply by PaulKappmeyer** [reply]
In my opinion, such feature is especially appropriate when the main purpose of the project is teaching.

I find it easier to understand (particularly larger) circuits when the layout is also visually appealing. I’ve attached a screenshot showing a very basic, but demonstrative example (coloring control wires in one and data wires in another color).
![wires with color](https://user-images.githubusercontent.com/42696082/131887233-03fdb506-d7f9-4974-910e-1b3967825b22.png)
compared to:
![wires without color](https://user-images.githubusercontent.com/42696082/131887745-4eb81fe9-cc60-4ce2-b791-25294c246e83.png)

Of course, this is just a very basic example, both designs are easy to understand and there are other ways to explain a circuit (instead of using colors). But larger circuits can get kind of messy and looking at just the basic blue wires can get confusing (and boring) too.

Still, I have to agree that this feature adds complexity for (technically seen) no functionality, and the risk of students getting lost when trying to make a circuit as colorful as possible is also there.
